File: android-badania-app/app/src/main/java/com/decentcode/badaniaapp/models/Badanie.kt
package com.decentcode.badaniaapp.models

import java.util.UUID

/**
 * Model danych badania lekarskiego
 */
data class Badanie(
    val id: String = UUID.randomUUID().toString(),
    val kod: String,
    val nazwa: String,
    val kwota: Double?
) {
    /**
     * Formatuje kwotę do wyświetlania w formacie "X,XX zł"
     */
    fun formattedKwota(): String {
        return kwota?.let {
            String.format("%.2f", it).replace(".", ",") + " zł"
        } ?: "-"
    }
}

/**
 * Reprezentuje wybrane badanie z ilością
 */
data class WybraneBadanie(
    val id: String = UUID.randomUUID().toString(),
    val badanie: Badanie,
    var ilosc: Int = 1
) {
    /**
     * Oblicza całkowitą kwotę dla tego badania (kwota * ilość)
     */
    fun calkowitaKwota(): Double {
        return badanie.kwota?.let { it * ilosc } ?: 0.0
    }

    /**
     * Formatuje całkowitą kwotę do wyświetlania
     */
    fun formattedCalkowitaKwota(): String {
        return String.format("%.2f", calkowitaKwota()).replace(".", ",") + " zł"
    }
}

File: ios-badania-app/BadaniaApp/BadaniaApp/Models/Badanie.swift
//
//  Badanie.swift
//  BadaniaApp
//
//  Model danych badania lekarskiego
//

import Foundation

struct Badanie: Identifiable, Codable, Equatable {
    let id: UUID
    let kod: String
    let nazwa: String
    let kwota: Double?
    
    init(id: UUID = UUID(), kod: String, nazwa: String, kwota: Double?) {
        self.id = id
        self.kod = kod
        self.nazwa = nazwa
        self.kwota = kwota
    }
    
    /// Formatuje kwotę do wyświetlania w formacie "X,XX zł"
    var formattedKwota: String {
        guard let kwota = kwota else {
            return "-"
        }
        return String(format: "%.2f", kwota).replacingOccurrences(of: ".", with: ",") + " zł"
    }
}

/// Reprezentuje wybrane badanie z ilością
struct WybraneBadanie: Identifiable, Equatable {
    let id: UUID
    let badanie: Badanie
    var ilosc: Int
    
    init(id: UUID = UUID(), badanie: Badanie, ilosc: Int = 1) {
        self.id = id
        self.badanie = badanie
        self.ilosc = ilosc
    }
    
    /// Oblicza całkowitą kwotę dla tego badania (kwota * ilość)
    var calkowitaKwota: Double {
        guard let kwota = badanie.kwota else {
            return 0.0
        }
        return kwota * Double(ilosc)
    }
    
    /// Formatuje całkowitą kwotę do wyświetlania
    var formattedCalkowitaKwota: String {
        return String(format: "%.2f", calkowitaKwota).replacingOccurrences(of: ".", with: ",") + " zł"
    }
}

